packages/maskbook/src/extension/background-script/EthereumServices/providers/MetaMask.ts

```diff
@@ -21,9 +21,10 @@ async function onAccountsChanged(accounts: string[]) {
     currentIsMetamaskLockedSettings.value = !(await provider!._metamask?.isUnlocked()) && accounts.length === 0
 }
 
-function onChainIdChanged(id: string) {
+async function onChainIdChanged(id: string) {
     // learn more: https://docs.metamask.io/guide/ethereum-provider.html#chain-ids and https://chainid.network/
     const chainId = Number.parseInt(id, 16) as ChainId
+    currentIsMetamaskLockedSettings.value = !(await provider!._metamask?.isUnlocked())
```

Comment:
After locked, then if unlock the metamask from metamask app directly, then click the "connect metamask wallet" button would not trigger `onAccountsChanged`, but `onChainIdChanged` would still be triggered. So set `currentIsMetamaskLockedSettings` here.